pkg/services/policer/check.go

```diff
@@ -127,6 +127,13 @@ func (p *Policer) processObject(ctx context.Context, addrWithType objectcore.Add
 		p.processNodes(c, addrWithType, nn[i], policy.ReplicaNumberByIndex(i), checkedNodes)
 	}
 
+	// if context is done, needLocalCopy might not be able to calculate
```

Comment:
How about switching the default? Initializing `needLocalCopy` to `true` above (or switching to `redundantLocalCopy` with `false` default). So we'd have `c.redundantLocalCopy` only when we're sure it's redundant.

Reply:
Thought about that, but this is refactor, don't wanna mix it with the fix. Maybe I'll add it in after processing the main part.

Reply:
OK, I'm just thinking that this particular patch in some ways is necessary only because we have this default that is not safe.